modules/renter/contractor/editor.go

```diff
@@ -145,6 +151,21 @@ func (c *Contractor) Editor(contract modules.RenterContract) (Editor, error) {
 		}
 	}
 
+	// acquire revising lock
+	c.mu.Lock()
+	alreadyRevising := c.revising[contract.ID]
+	if alreadyRevising {
+		c.mu.Unlock()
+		return nil, errors.New("already revising that contract")
+	}
+	c.revising[contract.ID] = true
+	c.mu.Unlock()
+	releaseLock := func() {
+		c.mu.Lock()
+		delete(c.revising, contract.ID)
+		c.mu.Unlock()
+	}
```

Comment:
That's probably better. Have to make sure you are handling errors correctly, but you have to do that anyway.